engine/language-server/src/main/scala/org/enso/languageserver/boot/LanguageServerComponent.scala

```diff
@@ -156,13 +191,17 @@ object LanguageServerComponent {
     * @param sampler a sampler gathering the application performance statistics
     * @param mainModule a main module containing all components of the server
     * @param jsonBinding a http binding for rpc protocol
+    * @param secureJsonBinding an optional https binding for rpc protocol
     * @param binaryBinding a http binding for data protocol
+    * @param secureBinaryBinding an optional https binding for data protocol
     */
   case class ServerContext(
     sampler: MethodsSampler,
     mainModule: MainModule,
     jsonBinding: Http.ServerBinding,
-    binaryBinding: Http.ServerBinding
+    secureJsonBinding: Option[Http.ServerBinding],
+    binaryBinding: Http.ServerBinding,
+    secureBinaryBinding: Option[Http.ServerBinding]
```

Comment:
I thought that the language server is started using either a secure or insecure connection. So that it will be `type Binding = Secure ServerBinding | Insecure ServerBinding`? Otherwise, it seems like it can have two web socket connections per protocol.

Reply:
It's not `xor`, no. We were asked to support both, especially for local connections.